apps/libsignpost/signpost_api.h

```diff
@@ -261,23 +265,39 @@ int signpost_processing_reply(uint8_t src_addr, uint8_t message_type, uint8_t* r
 /* ENERGY API                                                             */
 /**************************************************************************/
 
+//message types
 enum energy_message_type {
     EnergyQueryMessage = 0,
     EnergyLevelWarning24hMessage = 1,
     EnergyLevelCritical24hMessage = 2,
     EnergyCurrentWarning60sMessage = 3,
+    EnergyReportMessage = 4,
+    EnergyDutyCycleMessage = 5,
 };
 
+//information sent to a module from the controller
 typedef struct __attribute__((packed)) energy_information {
-    uint32_t    energy_limit_24h_mJ;
-    uint32_t    energy_used_24h_mJ;
-    uint16_t    current_limit_60s_mA;
-    uint16_t    current_average_60s_mA;
+    uint32_t    energy_limit_mWh;
+    uint16_t    average_power_mW;
     uint8_t     energy_limit_warning_threshold;
     uint8_t     energy_limit_critical_threshold;
 } signpost_energy_information_t;
 
-_Static_assert(sizeof(signpost_energy_information_t) == 14, "On-wire structure size");
+_Static_assert(sizeof(signpost_energy_information_t) == 8, "On-wire structure size");
+
+//a mechanism for modules to report energy usage from other modules
+//this structure is for each module
+typedef struct __attribute__((packed)) energy_report_module {
+    uint8_t module_address; //module i2c address
+    uint8_t module_percent; //an integer percent 0-100 that  the module has used
```

Comment:
The window is since the last call. In practice that doesn't matter much because the radio just sends these periodically (null or not). 

I don't really understand what you are getting at with your second point at all. What's the difference between an array of percentages for all modules and not sending a report for a module when it has used none of the radio energy? Re: amortization of wakeup - it would be great to do and I'm open to suggestions of interface improvements, but I don't think those improvements should block this pull request.

In general these interfaces are not perfect. We could think up a whole slew of improvements, but they are dead simple and I have implementations for them that work pretty well right now. Something that works for many cases is better than nothing.